docs/api/standard_library_types.md

```diff
@@ -386,6 +386,412 @@ except ValidationError as e:
+from typing import Iterable
+
+from pydantic import BaseModel
+
+
+class Model(BaseModel):
+    infinite: Iterable[int]
+
+
+def infinite_ints():
+    i = 0
+    while True:
+        yield i
+        i += 1
+
+
+m = Model(infinite=infinite_ints())
+print(m)
+"""
+infinite=ValidatorIterator(index=0, schema=Some(Int(IntValidator { strict: false })))
+"""
+
+for i in m.infinite:
+    print(i)
+    #> 0
+    #> 1
+    #> 2
+    #> 3
+    #> 4
+    #> 5
+    #> 6
+    #> 7
+    #> 8
+    #> 9
+    #> 10
+    if i == 10:
+        break
+```
+
+!!! warning
+    During initial validation, `Iterable` fields only perform a simple check that the provided argument is iterable.
+    To prevent it from being consumed, no validation of the yielded values is performed eagerly.
+
+
+Though the yielded values are not validated eagerly, they are still validated when yielded, and will raise a
+`ValidationError` at yield time when appropriate:
+
+```python
+from typing import Iterable
+
+from pydantic import BaseModel, ValidationError
+
+
+class Model(BaseModel):
+    int_iterator: Iterable[int]
+
+
+def my_iterator():
+    yield 13
+    yield '27'
+    yield 'a'
+
+
+m = Model(int_iterator=my_iterator())
+print(next(m.int_iterator))
+#> 13
+print(next(m.int_iterator))
+#> 27
+try:
+    next(m.int_iterator)
+except ValidationError as e:
+    print(e)
+    """
+    1 validation error for ValidatorIterator
+    2
+      Input should be a valid integer, unable to parse string as an integer [type=int_parsing, input_value='a', input_type=str]
+    """
+```
+
+## Dicts and Mapping Types
+
+### `dict`
+
+`dict(v)` is used to attempt to convert a dictionary. see `typing.Dict` below for sub-type constraints.
+
+```py
+from pydantic import BaseModel, ValidationError
+
+
+class Model(BaseModel):
+    x: dict
+
+
+m = Model(x={'foo': 1})
+print(m.model_dump())
+#> {'x': {'foo': 1}}
+
+try:
+    Model(x='test')
+except ValidationError as e:
+    print(e)
+    """
+    1 validation error for Model
+    x
+      Input should be a valid dictionary [type=dict_type, input_value='test', input_type=str]
+    """
+```
+
+### `typing.Dict`
+
+```py
+from typing import Dict
+
+from pydantic import BaseModel, ValidationError
+
+
+class Model(BaseModel):
+    x: Dict[str, int]
+
+
+m = Model(x={'foo': 1})
+print(m.model_dump())
+#> {'x': {'foo': 1}}
+
+try:
+    Model(x={'foo': '1'})
+except ValidationError as e:
+    print(e)
+    """
+    1 validation error for Model
+    x
+      Input should be a valid dictionary [type=dict_type, input_value='test', input_type=str]
+    """
+```
+
+### TypedDict
+
+!!! note
+    This is a new feature of the Python standard library as of Python 3.8.
+    Because of limitations in `typing.TypedDict` before 3.12, the [typing-extensions](https://pypi.org/project/typing-extensions/)
+    package is required for Python <3.12. You'll need to import `TypedDict` from `typing_extensions` instead of `typing` and will
+    get a build time error if you don't.
```

Comment:
Is this still true in Pydantic V2?

Reply:
we will get a `pydantic.errors.PydanticUserError` if we user `typing.TypedDict` with Python < 3.12